lib/active_fedora/relation/delegation.rb

```diff
@@ -13,7 +13,8 @@ module Delegation # :nodoc:
       :keep_if, :pop, :shift, :delete_at, :select!
     ].to_set
 
-    delegate :length, :collect, :map, :each, :all?, :include?, :to_ary, to: :to_a
+    delegate :length, :map, :to_ary, to: :to_a
+    delegate :all?, :blank?, :collect, :include?, :present?, to: :each
```

Comment:
is it possible to add `first` to delegate list?

Reply:
I think `#first` has its own implementation at https://github.com/samvera/active_fedora/blob/305dcb29fe978b7eed08909f2f8a000add1e37e0/lib/active_fedora/relation/finder_methods.rb#L3-L13

Reply:
perhaps this could be refactored now to be basically a delegation. I'm not sure the `limit(1).to_a[0]` approach gets us anything over `#each.first`, and the `loaded?` branch is already addressed in `#each`, now.

I'm inclined to benchmark before tweaking this one.

Reply:
Intersting, you're right... and even though it has that `limit(1).to_a[0]` thing on it, I _think_ it is loading the whole association for me. It's hard to know what `limit(1)` does without lots of time trying to figure it out. 

Have you confirmed that your `enum_for(:find_each, where_values)` implementation _really is_ avoiding loading the whole thing at once?  I'm finding it very hard to predict what will or won't.

Reply:
yes, I've run benchmarks to confirm that this approach works.

If you're having this problem with an `Association`, rather than a raw relation, my guess is that something else is calling `#to_a` *before* you call `#first`, and prewarming the cache.

Reply:
@jrochkind see: https://github.com/samvera/active_fedora/blob/1df7e4506811a8d50d425e9dc9becec255b18758/lib/active_fedora/associations/collection_association.rb#L301-L306

Reply:
yes, thanks, I've been staring at that code myself over the past week trying to figure out what's going on. I still can't say I understand it, or what to do about it. But I did figure out that line was relevant somehow. :)